src/EnergyPlus/EMSManager.cc

```diff
@@ -94,6 +94,37 @@ namespace EMSManager {
     // Using/Aliasing
     using namespace DataRuntimeLanguage;
 
+    constexpr std::array<std::string_view, static_cast<int>(EMSCallFrom::Num)> EMSCallFromNamesUC{"ENDOFZONESIZING",
+                                                                                                  "ENDOFSYSTEMSIZING",
+                                                                                                  "BEGINNEWENVIRONMENT",
+                                                                                                  "AFTERNEWENVIRONMENTWARMUPISCOMPLETE",
+                                                                                                  "BEGINTIMESTEPBEFOREPREDICTOR",
+                                                                                                  "AFTERPREDICTORBEFOREHVACMANAGERS",
+                                                                                                  "AFTERPREDICTORAFTERHVACMANAGERS",
+                                                                                                  "INSIDEHVACSYSTEMITERATIONLOOP",
+                                                                                                  "ENDOFSYSTEMTIMESTEPBEFOREHVACREPORTING",
+                                                                                                  "ENDOFSYSTEMTIMESTEPAFTERHVACREPORTING",
+                                                                                                  "ENDOFZONETIMESTEPBEFOREZONEREPORTING",
+                                                                                                  "ENDOFZONETIMESTEPAFTERZONEREPORTING",
+                                                                                                  "SETUPSIMULATION",
+                                                                                                  "EXTERNALINTERFACE",
+                                                                                                  "AFTERCOMPONENTINPUTREADIN",
+                                                                                                  "USERDEFINEDCOMPONENTMODEL",
+                                                                                                  "UNITARYSYSTEMSIZING",
+                                                                                                  "BEGINZONETIMESTEPBEFOREINITHEATBALANCE",
+                                                                                                  "BEGINZONETIMESTEPAFTERINITHEATBALANCE",
+                                                                                                  "BEGINZONETIMESTEPBEFORESETCURRENTWEATHER"};
+
+    std::array<std::string, static_cast<int>(SPControlType::Num)> controlTypeName{"Temperature Setpoint",
```

Comment:
This should be a `std::string_view`. That will allow this entire array to be a compile time constant.